### PR TITLE
hal: renesas: rz: Initial support for I2C

### PR DESCRIPTION
**Pull request by nhutnguyenkc**
Initial HAL support for I2C

### CODE REVIEW

**Review by thaoluonguw — APPROVED** [review]
It looks good to me.

**Review by duynguyenxa — APPROVED** [review]

**Review by KhiemNguyenT — APPROVED** [review]

**Comment by KhiemNguyenT** [comment]
@nhutnguyenkc , to get it merged, please resolve the conflict.

File: zephyr/rz/rz_cfg/fsp_cfg/rzg/r_riic_master_cfg.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+*/
+#ifndef R_RIIC_MASTER_CFG_H_
+#define R_RIIC_MASTER_CFG_H_
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define RIIC_MASTER_CFG_PARAM_CHECKING_ENABLE (BSP_CFG_PARAM_CHECKING_ENABLE)
+#define RIIC_MASTER_CFG_DMAC_ENABLE (0)
+#define RIIC_MASTER_CFG_ADDR_MODE_10_BIT_ENABLE (0)
```
Comment (duynguyenxa) [inline zephyr/rz/rz_cfg/fsp_cfg/rzg/r_riic_master_cfg.h:14]: @nhutnguyenkc , is it for now I2C driver of RZ will not support the 10 bit addressing mode?

Comment (nhutnguyenkc) [reply zephyr/rz/rz_cfg/fsp_cfg/rzg/r_riic_master_cfg.h]: @duynguyenxa , thank you, I have enabled 10-bit addressing mode, and already passed i2c test.

**Comment by nhutnguyenkc** [comment]
@KhiemNguyenT , I have rebased to resolve the conflict. Please kindly check again and merge it.

**Review by KhiemNguyenT — APPROVED** [review]

**Pull request merged**